cpp/src/gandiva/bitmap_accumulator.cc

```diff
@@ -28,15 +28,15 @@ void BitMapAccumulator::ComputeResult(uint8_t* dst_bitmap) {
     // set all bits to 0.
     memset(dst_bitmap, 0, arrow::BitUtil::BytesForBits(num_records));
   } else {
-    IntersectBitMaps(dst_bitmap, src_maps_, num_records);
+    IntersectBitMaps(dst_bitmap, src_maps_, src_map_offsets_, num_records);
   }
 }
 
 /// Compute the intersection of multiple bitmaps.
 void BitMapAccumulator::IntersectBitMaps(uint8_t* dst_map,
```

Comment:
i think you'll need to make a change in bitmap_accumulator_test.cc also for the extra arg ? can you please add a test there for "buffers with offsets" - that will be a good unit test for this. ?